### DISCUSSION TITLE
Is that possible to avoid compilation time if we re-solve the problem (with parameter value changed) multiple time?

### DISCUSSION THREAD

**Discussion by wadmes** [cvxpy/cvxpy#1943] [Q&A]
Hi everyone,

I am using cvxpy to implement an iterative SDP-based method.

The code is done, and the compilation / solve runtime are approximately 50 s / 10 s per iteration

As stated in the title, I would like to know whether it is possible to avoid the compilation after the first compilation?

Thanks!

**Comment by SteveDiamond** [comment]
Yes, if you solve the same problem with parameter value changed it should only compile once. We also have code generation to make things even faster: https://github.com/cvxgrp/cvxpygen

**Reply by wadmes** [reply]
Thank you for the reply!

Is there a parameter to make it compile once? I set `verbose = True`, and it shows that the compilation time among different iterations is not changed.

Here is the iteration part of my code:

```
    while iteration < 10:
        print("---------------------Solve X, iter {}-------------------------".format(str(iteration)))
        # set MSK_IPAR_NUM_THREADS to max number of threads
        start = timeit.default_timer()
        if opt.solver == 'MOSEK':
            prob.solve(warm_start = True, solver='MOSEK',mosek_params = {"MSK_IPAR_NUM_THREADS":os.cpu_count()},verbose=True,ignore_dpp = True)
        elif opt.solver == 'SCS':
            prob.solve(warm_start = True, solver='SCS',verbose=True,ignore_dpp = True)
        else:
            # raise not implemented error
            raise NotImplementedError
        W.value, ZW = opt_W(Z.value,2,opt)
        # for each gamma, plot X coordinates (Z[0:2,2:])
        iteration += 1

```

**Reply by SteveDiamond** [reply]
If you do ``ignore_dpp= True`` then you've turned off parameters. Remove this flag.

**Reply by wadmes** [reply]
> If you do `ignore_dpp= True` then you've turned off parameters. Remove this flag.

Thank you for the reply and sorry that I missed the documentation about `ignore_dpp`. My parameter is actually a lager matrix with size $100\times 100$, therefore, it costs too much time (waited ~1 hour but still compiling, so I terminated it) if I do not set ignore_dpp to True. I guess I may look `[cvxpygen](https://github.com/cvxgrp/cvxpygen)`I want to speed up the program.

Another quick question: Does `ignore_dpp = True` also influence `warm_start`? Or MOSEK does not support `warm_start`? (https://stackoverflow.com/questions/45724387/how-can-i-solve-the-conic-optimization-by-mosek-with-hot-start)

Thank you again!

**Reply by SteveDiamond** [reply]
Currently the cvxpy-mosek interface does not support warm start.

**Reply by wadmes** [reply]
Thank you for the help!

For warm start, may I know the supported solver in cvxpy regarding SDP?

I have also tried  SCS, but it did not work (at least according to the solve time)

(I disable dpp, and update parameters at each iteration, is this the reason for not-working?)

Thanks!
